blocks/bricks/sequences.py

```diff
@@ -107,11 +112,15 @@ class MLP(Sequence, Initializable, Feedforward):
 
     """
     @lazy(allocation=['dims'])
-    def __init__(self, activations, dims, **kwargs):
+    def __init__(self, activations, dims, prototype=None, **kwargs):
         self.activations = activations
-
-        self.linear_transformations = [Linear(name='linear_{}'.format(i))
-                                       for i in range(len(activations))]
+        self.prototype = Linear() if prototype is None else prototype
+        self.linear_transformations = []
+        for i in range(len(activations)):
+            linear = copy.deepcopy(self.prototype)
```

Comment:
I was just aping what the `parallel` bricks do, for the sake of consistency.

Personally I like the idea of *factories* more than prototypes (and if you need to partially configure a class, make your factory `functools.partial(TheClass, some_keyword=this, some_other_keyword=that)`).